### PR TITLE
Improvement/1767 build component one by one

### LINKED ISSUE #1767: Build Debian repository one by one
**Component**: build, entrypoint

**Why this is needed**: actually, we build all Debian components for repository at once. We need to build them one by one to implement it in the buildchain.

**What should be done**: Modify the script to take a component as an argument, add it in the repository and build the next one.

**Implementation proposal** (strongly recommended): modify the function `buildrepo` 

**Test plan**:
Launch the script following the procedure in the PR #1732  but calling the script with only one component.

### PR DESCRIPTION
**Pull request by ycointe**
**Component**: build, entrypoint

**Context**:  MetalK8s on Ubuntu

**Summary**: We need to build several Debian repository and not a big one with several component.

**Acceptance criteria**: have several repositories built

1. Download the packages: `./doit.sh _download_deb_packages`
2. Move them to the right location (will be fixed in the next PR): `mv _build/root/packages/debian/* _build/packages/debian`
3. Create the directories for the repositories (in the next PR it will be done by the buildchain): `mkdir _build/root/packages/debian/{metalk8s-bionic,metalk8s-bionic-backports,metalk8s-bionic-updates,metalk8s-kubernetes-xenial,metalk8s-salt_ubuntu1804}`
4. Run the following shell command (in the next PR, this loop will be integrated in the buildchain):

```shell
for REPO in metalk8s-bionic  metalk8s-bionic-backports  metalk8s-bionic-updates  metalk8s-kubernetes-xenial  metalk8s-salt_ubuntu1804
do
docker run --rm -it                                                      \
  -v "$(pwd)/packages/debian/entrypoint.sh":/entrypoint.sh:ro            \
  -v "$(pwd)/_build/packages/debian/${REPO}":/packages:ro                \
  -v "$(pwd)/packages/debian/distributions":/distributions:ro            \
  -v "$(pwd)/_build/root/packages/debian/${REPO}":/repository            \
  --env TARGET_UID="$(id -u)" --env TARGET_GID="$(id -g)"                \
  metalk8s-deb-build:latest                                              \
  /entrypoint.sh buildrepo "$REPO"
done
```

5. Check that the repositories were created: `tree _build/root/packages/debian/`

6. Check that the repositories are working:

```
% docker run --rm -it -v $(pwd)/_build/root/packages/debian/:/repositories:ro metalk8s-deb-build:latest
# cat ->> /etc/apt/sources.list
deb [trusted=yes] file:///repositories/metalk8s-bionic bionic metalk8s-bionic
deb [trusted=yes] file:///repositories/metalk8s-bionic-backports bionic metalk8s-bionic-backports
deb [trusted=yes] file:///repositories/metalk8s-bionic-updates bionic metalk8s-bionic-updates
deb [trusted=yes] file:///repositories/metalk8s-kubernetes-xenial bionic metalk8s-kubernetes-xenial
deb [trusted=yes] file:///repositories/metalk8s-salt_ubuntu1804 bionic metalk8s-salt_ubuntu1804
# apt update
# apt install kubelet
```

---

Closes: #1767

### CODE REVIEW

File: packages/debian/entrypoint.sh [hunk 1]
```diff
@@ -38,15 +38,15 @@ rpm2deb() {
...
-    done
+
+    local -r reponame="$1"
+    local -r repodir="/metalk8s-deb/$1"
+
+    mkdir -p /repository/{conf,incoming}
+    cp /metalk8s-deb/distributions /repository/conf
+    sed -i 's/_REPONAME_/'"${reponame}"'/g' /repository/conf/distributions
+    reprepro -b /repository/ export
+    reprepro -C "$reponame" -b /repository/ includedeb bionic "$repodir"/*.deb
```
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: Missing `    chown -R "${TARGET_UID}:${TARGET_GID}" /repository`

File: packages/debian/entrypoint.sh [hunk 2]
```diff
@@ -38,15 +38,15 @@ rpm2deb() {
...
-    reprepro -b /repository/debian export
-    for dir in /metalk8s-deb/metalk8s-*; do
-        reprepro -C "${dir##*/}" -b /repository/debian \
-            includedeb bionic "$dir"/*.deb
-    done
+
+    local -r reponame="$1"
+    local -r repodir="/metalk8s-deb/$1"
+
+    mkdir -p /repository/{conf,incoming}
```
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: Remove `incoming` (seems unused) and remove `-p` (you know that `/repository` exists).

=> `mkdir /repository/conf`

File: packages/debian/entrypoint.sh [hunk 3]
```diff
@@ -38,15 +38,15 @@ rpm2deb() {
...
-    cp /metalk8s-deb/distributions /repository/debian/conf
-    #TODO: Mount `distributions` file in  /repository/debian/conf
-    reprepro -b /repository/debian export
-    for dir in /metalk8s-deb/metalk8s-*; do
-        reprepro -C "${dir##*/}" -b /repository/debian \
-            includedeb bionic "$dir"/*.deb
-    done
+
+    local -r reponame="$1"
+    local -r repodir="/metalk8s-deb/$1"
```
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: I don't think we need that one if we mount the as `  -v "$(pwd)/_build/packages/debian/${REPO}":/packages:ro`

File: packages/debian/entrypoint.sh [hunk 4]
```diff
@@ -38,15 +38,15 @@ rpm2deb() {
...
+
+    mkdir -p /repository/{conf,incoming}
+    cp /metalk8s-deb/distributions /repository/conf
```
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: `cp /distributions /repository/conf`

That way we can mount `/metalk8s-deb` (or `/packages` if you follow my previous comment) as read-only.

File: packages/debian/entrypoint.sh (hunk 1 above)
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: If we get rid of `$repodir` as suggested, simply write `"/packages/*.deb` here

**Review by slaperche-scality — CHANGES_REQUESTED** [review]

**Review by jbertran — DISMISSED** [review]

File: packages/debian/entrypoint.sh [hunk 5]
```diff
@@ -38,15 +38,15 @@ rpm2deb() {
...
-            includedeb bionic "$dir"/*.deb
-    done
+
+    local -r reponame="$1"
+
+    mkdir /repository/conf
+    cp /distributions /repository/conf
+    sed -i 's/_REPONAME_/'"${reponame}"'/g' /repository/conf/distributions
+    reprepro -b /repository/ export
+    reprepro -C "$reponame" -b /repository/ includedeb bionic "/packages"/*.deb
```
Comment (slaperche-scality) [inline packages/debian/entrypoint.sh]: `"/packages"/*.deb` => `/packages/*.deb`
No need to quote around a literal string.

**Review by jbertran — APPROVED** [review]

**Review by slaperche-scality — APPROVED** [review]

**Pull request merged**